src/nvim/api/ui_events.in.h

```diff
@@ -58,7 +58,8 @@ void set_title(String title)
   FUNC_API_SINCE(3);
 void set_icon(String icon)
   FUNC_API_SINCE(3);
-
+void highlight_info_set(Array highlights_changed)
+  FUNC_API_SINCE(3);
```

Comment:
I agree but let's keep that for another PR. Meanwhile I will extract the "highlight changed" into separate commit so that you can choose to include it or not (right now all UIs get notifications even when not interested).